stats/opentelemetry/client_metrics.go

```diff
@@ -163,15 +210,17 @@ func (h *clientStatsHandler) TagRPC(ctx context.Context, info *stats.RPCTagInfo)
 		}
 		ctx = istats.SetLabels(ctx, labels)
 	}
-	ai := &attemptInfo{ // populates information about RPC start.
-		startTime: time.Now(),
-		xdsLabels: labels.TelemetryLabels,
-		method:    info.FullMethodName,
+	ai := &attemptInfo{}
+	startTime := time.Now()
+	if !isTracingDisabled(h.options.TraceOptions) {
+		ctx, ai = h.traceTagRPC(ctx, info)
 	}
-	ri := &rpcInfo{
+	ai.startTime = startTime
```

Comment:
Why did you change how this is set?

Why are you setting a local 3 lines above to time.Now() and then assigning this field to that?

Reply:
Done